src/target/lpc546xx.c

```diff
@@ -78,6 +78,48 @@ const command_s lpc546xx_cmd_list[] = {
 	{NULL, NULL, NULL},
 };
 
+#define USE_LUT
+
+#ifdef USE_LUT
+struct lpc546xx_device {
+	uint32_t chipid;
+	const char *designator;
+	uint16_t flash_kbytes;
+	uint16_t sram123_kbytes;
+};
+
+/* Reference: "LPC546XX Product data sheet" revision 2.6, 2018
+ * Part type number encoding: LPC546xxJyyy, where yyy is flash size, KiB
+ */
+static const struct lpc546xx_device lpc546xx_devices_lut[] = {
+	{0x7f954605U, "LPC54605J256", 256, 32},
+	{0x7f954606U, "LPC54606J256", 256, 32},
+	{0x7f954607U, "LPC54607J256", 256, 32},
+	{0x7f954616U, "LPC54616J256", 256, 32},
+	{0xfff54605U, "LPC54605J512", 512, 96},
+	{0xfff54606U, "LPC54606J512", 512, 96},
+	{0xfff54607U, "LPC54607J512", 512, 96},
+	{0xfff54608U, "LPC54608J512", 512, 96},
+	{0xfff54616U, "LPC54616J512", 512, 96},
+	{0xfff54618U, "LPC54618J512", 512, 96},
+	{0xfff54628U, "LPC54628J512", 512, 96},
+};
+
+/* Look up device parameters */
+static const struct lpc546xx_device *lpc546xx_get_device(const uint32_t chipid)
+{
+	/* Linear search through chips */
+	for (size_t i = 0; i < (sizeof(lpc546xx_devices_lut) / sizeof(struct lpc546xx_device)); i++) {
+		if (lpc546xx_devices_lut[i].chipid == chipid) {
```

Comment:
The braces on the if block here are unnecessary, please remove them.

Reply:
Removed inner braces.